staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -93,13 +93,13 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 		return nil, err
 	}
 	// The options didn't require a custom TLS config
-	if tlsConfig == nil && config.Dial == nil && config.Proxy == nil {
+	if tlsConfig == nil && config.DialHolder == nil && config.Proxy == nil {
 		return http.DefaultTransport, nil
 	}
 
 	var dial func(ctx context.Context, network, address string) (net.Conn, error)
-	if config.Dial != nil {
-		dial = config.Dial
+	if config.DialHolder != nil {
+		dial = config.DialHolder.Dial
```

Comment:
can we rely on DialHolder.Dial being non-nil here?

Reply:
Yes, this one is safe.